lib/iris/util.py

```diff
@@ -2059,31 +2059,53 @@ def equalise_attributes(cubes):
     See more at :doc:`/userguide/real_and_lazy_data`.
 
     """
+    # deferred imports to avoid circularity problem
+    from iris.common._split_attribute_dicts import (
+        _convert_splitattrs_to_pairedkeys_dict,
+    )
+    from iris.cube import CubeAttrsDict
+
+    cube_attrs = [cube.attributes for cube in cubes]
     removed = []
+
+    # *IF* they are cube attributes, then convert to "paired-keys" form.
```

Comment:
Can this comment be enhanced to explain why the rest of the code 'magically works' once the keys are paired keys?

Reply:
I've tried now https://github.com/SciTools/iris/pull/5586/commits/c3e2175b7ea7735ed8ab2b49d926d70922a4afde
Please see what you think !